boa/integrations/jupyter/browser.py

```diff
@@ -224,9 +211,31 @@ def _parse_js_result(result: dict) -> Any:
     if "data" in result:
         return result["data"]
 
+    def _find_key(input_dict, target_key, typ) -> Any:
+        for key, value in input_dict.items():
+            if isinstance(value, dict):
+                found = _find_key(value, target_key, typ)
+                if found is not None:
+                    return found
+            if key == target_key and isinstance(value, typ) and value != "error":
```

Comment:
what happens when you call `_find_key(error, "error", str)` a few lines below?

Reply:
ah the check is for `value == "error"`, not `key == "error"`. what is the purpose of the check then?

Reply:
The RPCs return inconsistent errors. This code is trying to find the best error to describe the situation.
For example, this test returns `error.value="error"`:
https://github.com/vyperlang/titanoboa/blob/2dfdf53fff6584504c8ecb6ce7712d3062aa1d3d/tests/unitary/jupyter/test_browser.py#L189